code/datums/black_market_item.dm

```diff
@@ -72,62 +74,118 @@ var/list/black_market_items = list()
 	..()
-	if(!istype(U)) 
+	if(!istype(radio)) 
 		return 0
-
 	if(user.stat || user.restrained())
 		return 0
-
 	if(!(istype(user,/mob/living/carbon/human)))
 		return 0
-
 	if(get_stock() <= 0)
 		to_chat(user, "<span class='warning'>This item is out of stock. Scram.</span>")
 		return 0
+	if(!((radio.loc in user.contents) || (in_range(radio.loc, user) && istype(radio.loc.loc, /turf))))
+		return 0
+	if(get_cost() > radio.money_stored)
+		return 0
 
-	// If the black_market's holder is in the user's content
-	if((U.loc in user.contents) || (in_range(U.loc, user) && istype(U.loc.loc, /turf)))
-		if(get_cost() > U.money_stored)
-			return 0
+	if(delivery_method == 1)
+		spawn_cheap(radio,user)
+	else if(delivery_method == 2)
+		spawn_normal(radio,user)
+	else
+		spawn_expensive(radio,user)
 
-		var/obj/I = spawn_item(get_turf(user), U, user)
-		if(!I)
+/datum/black_market_item/proc/spawn_cheap(var/obj/item/device/illegalradio/radio, var/mob/user)
+	var/direction = pick(list(NORTH,EAST,SOUTH,WEST))
+	var/direction_string
+	if(direction == NORTH) //Look, I'm not sure why directions are 1/2/4/8 and why arrays start at 1 and why I can't just do direction/2+1 to get the index of an array because integer division is fucked or something with nonstatic typing, just let it be beautiful gunk :)
+		direction_string = "north"
+	else if(direction == EAST)
+		direction_string = "east"
+	else if(direction == SOUTH)
+		direction_string = "south"
+	else if(direction == WEST)
+		direction_string = "west"
+	log_transaction("The item was launched at the station from the [direction_string].", user)
+	radio.visible_message("The uplink beeps: <span class='warning'>Your item was launched from the [direction_string]. It will impact the station in less than a minute.</span>")
+	spawn(rand(150,450))
+		var/obj/item = spawn_item(get_turf(user), radio, user)
+		if(!item)
 			return 0
-		after_spawn(I,user)
-		var/icon/tempimage = icon(I.icon, I.icon_state)
-		end_icons += tempimage
-		var/tempstate = end_icons.len
-
-		if(ishuman(user))
-			var/mob/living/carbon/human/A = user
-
-			if(istype(I, /obj/item))
-				A.put_in_any_hand_if_possible(I)
-
-			U.purchase_log += {"[user] ([user.ckey]) bought <img src="logo_[tempstate].png"> [name] for [get_cost()]."}
-			//stat_collection.black_market_purchase(src, I, user)
-			if(get_stock() != 0)
-				round_stock -= 1
+		after_spawn(item,1,user)
+		item.ThrowAtStation(30,2,direction)
+		if(get_stock() != 0)
+			round_stock -= 1	
+		spawn(rand(300,600))
+			if(!radio.nanotrasen_variant && prob(sps_chances[1]))
+				SPS_black_market_alert(src, "The SPS decryption complex has detected an illegal black market purchase of item [name]. It was launched at the station recently.")
+	
+/datum/black_market_item/proc/spawn_normal(var/obj/item/device/illegalradio/radio, var/mob/user)
+	var/list/potential_locations = list()
+	var/turf/spawnloc = null
```

Comment:
You don't actually have to set this as null, that's the default value. Just `var/turf/spawnloc` is fine